### ISSUE TITLE
HA entity state for HVAC reverts back to previous value

### ISSUE THREAD

**Issue by andjo** [alandtse/tesla#226]
## Version of the custom_component

v2.3.0

## Configuration

```yaml
service: climate.set_hvac_mode
data:
  hvac_mode: heat_cool
target:
  entity_id: climate.teslarossa_hvac_climate_system
```

## Describe the bug

Turning HVAC on or off sets the entity state to the new value for a short time (about 0-9 seconds) but then reverts back to previous state. The command works on the car, it's just the state in Home Assistant that is out of synch with reality.

## Debug log

```text
2022-06-12 21:53:06 DEBUG (MainThread) [custom_components.tesla_custom] Running controller.update()
2022-06-12 21:53:06 DEBUG (MainThread) [teslajsonpy.controller] Get vehicles. Force: False Time: 30 Interval 60
2022-06-12 21:53:06 DEBUG (MainThread) [teslajsonpy.controller] 16553: online. Polling policy: normal. Update state: normal. Since last park: 99. Since last wake_up: 97. Idle interval: 600. shift_state: None sentry: False climate: False, charging: Complete
2022-06-12 21:53:06 DEBUG (MainThread) [teslajsonpy.controller] 16553: Skipping update with state online. Polling: True. Last update: 96 ago. Last parked: 99 ago. Last wake_up 97 ago.
2022-06-12 21:53:06 DEBUG (MainThread) [custom_components.tesla_custom] Finished fetching tesla_custom data in 0.002 seconds (success: True)
2022-06-12 21:53:14 DEBUG (MainThread) [custom_components.tesla_custom.climate] Teslarossa HVAC (climate) system: Setting hvac mode to heat_cool
2022-06-12 21:53:14 DEBUG (MainThread) [teslajsonpy.connection] Token expiration in 4:09:56
2022-06-12 21:53:14 DEBUG (MainThread) [teslajsonpy.connection] post: https://owner-api.teslamotors.com/api/1/vehicles/<redacted>/command/auto_conditioning_start {}
2022-06-12 21:53:15 DEBUG (MainThread) [teslajsonpy.connection] 200: {"response":{"reason":"","result":true}}
2022-06-12 21:53:15 DEBUG (MainThread) [teslajsonpy.controller] Get vehicles. Force: False Time: 39 Interval 60
2022-06-12 21:53:15 DEBUG (MainThread) [teslajsonpy.controller] 16553: online. Polling policy: normal. Update state: normal. Since last park: 108. Since last wake_up: 106. Idle interval: 600. shift_state: None sentry: False climate: False, charging: Complete
2022-06-12 21:53:15 DEBUG (MainThread) [teslajsonpy.controller] 16553: Skipping update with state online. Polling: True. Last update: 105 ago. Last parked: 108 ago. Last wake_up 106 ago.
2022-06-12 21:53:15 DEBUG (MainThread) [teslajsonpy.controller] Get vehicles. Force: True Time: 39 Interval 60
2022-06-12 21:53:15 DEBUG (MainThread) [teslajsonpy.connection] Token expiration in 4:09:55
2022-06-12 21:53:15 DEBUG (MainThread) [teslajsonpy.connection] get: https://owner-api.teslamotors.com/api/1/vehicles {}
2022-06-12 21:53:15 DEBUG (MainThread) [teslajsonpy.connection] 200: {"response":[<redacted>],"count":1}
2022-06-12 21:53:15 DEBUG (MainThread) [teslajsonpy.controller] 16553: Updating VEHICLE_DATA
2022-06-12 21:53:15 DEBUG (MainThread) [teslajsonpy.connection] Token expiration in 4:09:55
2022-06-12 21:53:15 DEBUG (MainThread) [teslajsonpy.connection] get: https://owner-api.teslamotors.com/api/1/vehicles/<redacted>/vehicle_data {}
2022-06-12 21:53:15 DEBUG (MainThread) [teslajsonpy.connection] 200: {"response":{<redacted>}}
2022-06-12 21:53:15 DEBUG (MainThread) [custom_components.tesla_custom.climate] Refreshing Device: switch.heated steering switch
2022-06-12 21:53:15 DEBUG (MainThread) [custom_components.tesla_custom.climate] Refreshing Device: select.heated seat left
2022-06-12 21:53:15 DEBUG (MainThread) [custom_components.tesla_custom.climate] Refreshing Device: select.heated seat right
2022-06-12 21:53:15 DEBUG (MainThread) [custom_components.tesla_custom.climate] Refreshing Device: select.heated seat rear_left
2022-06-12 21:53:15 DEBUG (MainThread) [custom_components.tesla_custom.climate] Refreshing Device: select.heated seat rear_center
2022-06-12 21:53:15 DEBUG (MainThread) [custom_components.tesla_custom.climate] Refreshing Device: select.heated seat rear_right
2022-06-12 21:53:15 DEBUG (MainThread) [custom_components.tesla_custom.climate] Refreshing Device: select.heated seat third_row_left
2022-06-12 21:53:15 DEBUG (MainThread) [custom_components.tesla_custom.climate] Refreshing Device: select.heated seat third_row_right
2022-06-12 21:53:16 DEBUG (MainThread) [custom_components.tesla_custom] Running controller.update()
2022-06-12 21:53:16 DEBUG (MainThread) [teslajsonpy.controller] Get vehicles. Force: False Time: 1 Interval 60
2022-06-12 21:53:16 DEBUG (MainThread) [teslajsonpy.controller] 16553: online. Polling policy: normal. Update state: normal. Since last park: 109. Since last wake_up: 107. Idle interval: 600. shift_state: None sentry: False climate: False, charging: Complete
2022-06-12 21:53:16 DEBUG (MainThread) [teslajsonpy.controller] 16553: Skipping update with state online. Polling: True. Last update: 0 ago. Last parked: 109 ago. Last wake_up 107 ago.
2022-06-12 21:53:16 DEBUG (MainThread) [custom_components.tesla_custom] Finished fetching tesla_custom data in 0.002 seconds (success: True)
2022-06-12 21:53:26 DEBUG (MainThread) [custom_components.tesla_custom] Running controller.update()
2022-06-12 21:53:26 DEBUG (MainThread) [teslajsonpy.controller] Get vehicles. Force: False Time: 11 Interval 60
2022-06-12 21:53:26 DEBUG (MainThread) [teslajsonpy.controller] 16553: online. Polling policy: normal. Update state: normal. Since last park: 119. Since last wake_up: 117. Idle interval: 600. shift_state: None sentry: False climate: False, charging: Complete
2022-06-12 21:53:26 DEBUG (MainThread) [teslajsonpy.controller] 16553: Skipping update with state online. Polling: True. Last update: 10 ago. Last parked: 119 ago. Last wake_up 117 ago.
2022-06-12 21:53:26 DEBUG (MainThread) [custom_components.tesla_custom] Finished fetching tesla_custom data in 0.003 seconds (success: True)
```

**Comment by hazza64** [comment]
I think this may be due to a new Tesla firmware update as this started happening to me only once my 3 got on 2022.12.3.16. They added a new feature to do with HVAC which may have broken that functionality; they added the ability of turning on Dog Mode and Camp Mode to the app.

**Comment by alandtse** [comment]
Unfortunately, you've redacted the information from the logs showing the HVAC state so it's not helpful. Please provide a log showing that state at least.

This line needs to keep at least the hvac info intact.
```
2022-06-12 21:53:15 DEBUG (MainThread) [teslajsonpy.connection] 200: {"response":{<redacted>}}
```

**Comment by andjo** [comment]
Oh, sorry about that! Didn't want to publish VIN and tokens and stuff. Did it more selectively this time.
This is me turning climate on at about `21:02:07`:
```
2022-06-19 21:02:04 DEBUG (MainThread) [custom_components.tesla_custom] Running controller.update()
2022-06-19 21:02:04 DEBUG (MainThread) [teslajsonpy.controller] Get vehicles. Force: False Time: 29 Interval 60
2022-06-19 21:02:04 DEBUG (MainThread) [teslajsonpy.controller] 16553: online. Polling policy: normal. Update state: normal. Since last park: 143. Since last wake_up: 141. Idle interval: 600. shift_state: None sentry: False climate: False, charging: Stopped
2022-06-19 21:02:04 DEBUG (MainThread) [teslajsonpy.controller] 16553: Skipping update with state online. Polling: True. Last update: 28 ago. Last parked: 143 ago. Last wake_up 141 ago.
2022-06-19 21:02:04 DEBUG (MainThread) [custom_components.tesla_custom] Finished fetching tesla_custom data in 0.002 seconds (success: True)
2022-06-19 21:02:07 DEBUG (MainThread) [custom_components.tesla_custom.climate] Teslarossa HVAC (climate) system: Setting hvac mode to heat_cool
2022-06-19 21:02:07 DEBUG (MainThread) [teslajsonpy.connection] Token expiration in 5:05:49
2022-06-19 21:02:07 DEBUG (MainThread) [teslajsonpy.connection] post: https://owner-api.teslamotors.com/api/1/vehicles/1492932625928581/command/auto_conditioning_start {}
2022-06-19 21:02:08 DEBUG (MainThread) [teslajsonpy.connection] 200: {"response":{"reason":"","result":true}}
2022-06-19 21:02:08 DEBUG (MainThread) [teslajsonpy.controller] Get vehicles. Force: False Time: 33 Interval 60
2022-06-19 21:02:08 DEBUG (MainThread) [teslajsonpy.controller] 16553: online. Polling policy: normal. Update state: normal. Since last park: 147. Since last wake_up: 145. Idle interval: 600. shift_state: None sentry: False climate: False, charging: Stopped
2022-06-19 21:02:08 DEBUG (MainThread) [teslajsonpy.controller] 16553: Skipping update with state online. Polling: True. Last update: 32 ago. Last parked: 147 ago. Last wake_up 145 ago.
2022-06-19 21:02:08 DEBUG (MainThread) [teslajsonpy.controller] Get vehicles. Force: True Time: 33 Interval 60
2022-06-19 21:02:08 DEBUG (MainThread) [teslajsonpy.connection] Token expiration in 5:05:48
2022-06-19 21:02:08 DEBUG (MainThread) [teslajsonpy.connection] get: https://owner-api.teslamotors.com/api/1/vehicles {}
2022-06-19 21:02:08 DEBUG (MainThread) [teslajsonpy.connection] 200: {"response":[{"id":123,"vehicle_id":123,"vin":"123","display_name":"Teslarossa","option_codes":"AD15,AF00,APF2,APH3,AU3P,BC3B,BT37,CDM0,CH06,COSE,DRLH,DV4W,FC3P,FG31,FM3B,GLFR,HL31,HM31,ID3W,IL31,LTPB,MDL3,MR31,PPMR,PC30,REEU,RF3G,RS3H,S3PB,SA3P,SC04,STCP,SU3C,T3MS,TM00,TW00,UT3P,W38B,WR00,ZINV,MI00,PL30,SLR0,ST31,BG31,I36M,OSSB,RSF1,CPF1","color":null,"access_type":"OWNER","tokens":["123","123"],"state":"online","in_service":false,"id_s":"123","calendar_enabled":true,"api_version":40,"backseat_token":null,"backseat_token_updated_at":null}],"count":1}
2022-06-19 21:02:08 DEBUG (MainThread) [teslajsonpy.controller] 16553: Updating VEHICLE_DATA
2022-06-19 21:02:08 DEBUG (MainThread) [teslajsonpy.connection] Token expiration in 5:05:48
2022-06-19 21:02:08 DEBUG (MainThread) [teslajsonpy.connection] get: https://owner-api.teslamotors.com/api/1/vehicles/123/vehicle_data {}
2022-06-19 21:02:08 DEBUG (MainThread) [teslajsonpy.connection] 200: {"response":{"id":123,"user_id":123,"vehicle_id":123,"vin":"123","display_name":"Teslarossa","option_codes":"AD15,AF00,APF2,APH3,AU3P,BC3B,BT37,CDM0,CH06,COSE,DRLH,DV4W,FC3P,FG31,FM3B,GLFR,HL31,HM31,ID3W,IL31,LTPB,MDL3,MR31,PPMR,PC30,REEU,RF3G,RS3H,S3PB,SA3P,SC04,STCP,SU3C,T3MS,TM00,TW00,UT3P,W38B,WR00,ZINV,MI00,PL30,SLR0,ST31,BG31,I36M,OSSB,RSF1,CPF1","color":null,"access_type":"OWNER","tokens":["123","123"],"state":"online","in_service":false,"id_s":"123","calendar_enabled":true,"api_version":40,"backseat_token":null,"backseat_token_updated_at":null,"charge_state":{"battery_heater_on":false,"battery_level":49,"battery_range":144.39,"charge_amps":16,"charge_current_request":16,"charge_current_request_max":16,"charge_enable_request":false,"charge_energy_added":2.62,"charge_limit_soc":50,"charge_limit_soc_max":100,"charge_limit_soc_min":50,"charge_limit_soc_std":90,"charge_miles_added_ideal":10.5,"charge_miles_added_rated":10.5,"charge_port_cold_weather_mode":false,"charge_port_color":"<invalid>","charge_port_door_open":true,"charge_port_latch":"Engaged","charge_rate":0.0,"charge_to_max_range":false,"charger_actual_current":2,"charger_phases":2,"charger_pilot_current":16,"charger_power":1,"charger_voltage":233,"charging_state":"Stopped","conn_charge_cable":"IEC","est_battery_range":135.2,"fast_charger_brand":"<invalid>","fast_charger_present":false,"fast_charger_type":"<invalid>","ideal_battery_range":144.39,"managed_charging_active":false,"managed_charging_start_time":null,"managed_charging_user_canceled":false,"max_range_charge_counter":0,"minutes_to_full_charge":5,"not_enough_power_to_heat":null,"off_peak_charging_enabled":false,"off_peak_charging_times":"all_week","off_peak_hours_end_time":360,"preconditioning_enabled":false,"preconditioning_times":"all_week","scheduled_charging_mode":"StartAt","scheduled_charging_pending":true,"scheduled_charging_start_time":1655679600,"scheduled_charging_start_time_app":60,"scheduled_charging_start_time_minutes":60,"scheduled_departure_time":1609311600,"scheduled_departure_time_minutes":480,"supercharger_session_trip_planner":false,"time_to_full_charge":0.08,"timestamp":1655665328349,"trip_charging":false,"usable_battery_level":49,"user_charge_enable_request":null},"climate_state":{"allow_cabin_overheat_protection":false,"auto_seat_climate_left":true,"auto_seat_climate_right":true,"battery_heater":false,"battery_heater_no_power":null,"cabin_overheat_protection":"On","cabin_overheat_protection_actively_cooling":false,"climate_keeper_mode":"off","defrost_mode":0,"driver_temp_setting":21.0,"fan_status":0,"hvac_auto_request":"On","inside_temp":21.3,"is_auto_conditioning_on":false,"is_climate_on":false,"is_front_defroster_on":false,"is_preconditioning":true,"is_rear_defroster_on":false,"left_temp_direction":0,"max_avail_temp":28.0,"min_avail_temp":15.0,"outside_temp":23.0,"passenger_temp_setting":21.0,"remote_heater_control_enabled":true,"right_temp_direction":0,"seat_heater_left":0,"seat_heater_rear_center":0,"seat_heater_rear_left":0,"seat_heater_rear_right":0,"seat_heater_right":0,"side_mirror_heaters":false,"supports_fan_only_cabin_overheat_protection":true,"timestamp":1655665328348,"wiper_blade_heater":false},"drive_state":{"gps_as_of":1655562356,"heading":172,"latitude":58.405489,"longitude":15.625203,"native_latitude":58.405489,"native_location_supported":1,"native_longitude":15.625203,"native_type":"wgs","power":0,"shift_state":null,"speed":null,"timestamp":1655665328349},"gui_settings":{"gui_24_hour_time":true,"gui_charge_rate_units":"kW","gui_distance_units":"km/hr","gui_range_display":"Rated","gui_temperature_units":"C","show_range_units":false,"timestamp":1655665328349},"vehicle_config":{"aux_park_lamps":"Eu","badge_version":0,"can_accept_navigation_requests":true,"can_actuate_trunks":true,"car_special_type":"base","car_type":"model3","charge_port_type":"CCS","dashcam_clip_save_supported":true,"default_charge_to_max":false,"driver_assist":"TeslaAP3","ece_restrictions":true,"efficiency_package":"Default","eu_vehicle":true,"exterior_color":"RedMulticoat","exterior_trim":"Chrome","exterior_trim_override":"Chrome","has_air_suspension":false,"has_ludicrous_mode":false,"has_seat_cooling":false,"headlamp_type":"Premium","interior_trim_type":"Black","key_version":2,"motorized_charge_port":true,"paint_color_override":"10,1,1,0.1,0.04","performance_package":"Base","plg":false,"pws":false,"rear_drive_unit":"PM216MOSFET","rear_seat_heaters":1,"rear_seat_type":0,"rhd":false,"roof_color":"RoofColorGlass","seat_type":null,"spoiler_type":"None","sun_roof_installed":null,"third_row_seats":"None","timestamp":1655665328349,"trim_badging":"74d","use_range_badging":true,"utc_offset":7200,"webcam_supported":false,"wheel_type":"Pinwheel18"},"vehicle_state":{"api_version":40,"autopark_state_v2":"standby","autopark_style":"dead_man","calendar_supported":true,"car_version":"2022.16.1.1 49e211fc4716","center_display_state":0,"dashcam_clip_save_available":false,"dashcam_state":"Unavailable","df":0,"dr":0,"fd_window":0,"feature_bitmask":"ff,0","fp_window":0,"ft":0,"homelink_device_count":0,"homelink_nearby":false,"is_user_present":false,"last_autopark_error":"no_error","locked":true,"media_state":{"remote_control_enabled":true},"notifications_supported":true,"odometer":23902.921828,"parsed_calendar_supported":true,"pf":0,"pr":0,"rd_window":0,"remote_start":false,"remote_start_enabled":true,"remote_start_supported":true,"rp_window":0,"rt":0,"santa_mode":0,"sentry_mode":false,"sentry_mode_available":true,"service_mode":false,"service_mode_plus":false,"smart_summon_available":true,"software_update":{"download_perc":0,"expected_duration_sec":2700,"install_perc":1,"status":"","version":" "},"speed_limit_mode":{"active":false,"current_limit_mph":55.923407,"max_limit_mph":90,"min_limit_mph":50.0,"pin_code_set":false},"summon_standby_mode_enabled":false,"timestamp":1655665328348,"tpms_pressure_fl":null,"tpms_pressure_fr":null,"tpms_pressure_rl":null,"tpms_pressure_rr":null,"valet_mode":false,"valet_pin_needed":true,"vehicle_name":"Teslarossa","vehicle_self_test_progress":0,"vehicle_self_test_requested":false,"webcam_available":false}}}
2022-06-19 21:02:08 DEBUG (MainThread) [custom_components.tesla_custom.climate] Refreshing Device: switch.heated steering switch
2022-06-19 21:02:08 DEBUG (MainThread) [custom_components.tesla_custom.climate] Refreshing Device: select.heated seat left
2022-06-19 21:02:08 DEBUG (MainThread) [custom_components.tesla_custom.climate] Refreshing Device: select.heated seat right
2022-06-19 21:02:08 DEBUG (MainThread) [custom_components.tesla_custom.climate] Refreshing Device: select.heated seat rear_left
2022-06-19 21:02:08 DEBUG (MainThread) [custom_components.tesla_custom.climate] Refreshing Device: select.heated seat rear_center
2022-06-19 21:02:08 DEBUG (MainThread) [custom_components.tesla_custom.climate] Refreshing Device: select.heated seat rear_right
2022-06-19 21:02:08 DEBUG (MainThread) [custom_components.tesla_custom.climate] Refreshing Device: select.heated seat third_row_left
2022-06-19 21:02:08 DEBUG (MainThread) [custom_components.tesla_custom.climate] Refreshing Device: select.heated seat third_row_right
2022-06-19 21:02:14 DEBUG (MainThread) [custom_components.tesla_custom] Running controller.update()
2022-06-19 21:02:14 DEBUG (MainThread) [teslajsonpy.controller] Get vehicles. Force: False Time: 6 Interval 60
2022-06-19 21:02:14 DEBUG (MainThread) [teslajsonpy.controller] 16553: online. Polling policy: normal. Update state: normal. Since last park: 153. Since last wake_up: 151. Idle interval: 600. shift_state: None sentry: False climate: False, charging: Stopped
2022-06-19 21:02:14 DEBUG (MainThread) [teslajsonpy.controller] 16553: Skipping update with state online. Polling: True. Last update: 5 ago. Last parked: 153 ago. Last wake_up 151 ago.
2022-06-19 21:02:14 DEBUG (MainThread) [custom_components.tesla_custom] Finished fetching tesla_custom data in 0.002 seconds (success: True)
2022-06-19 21:02:24 DEBUG (MainThread) [custom_components.tesla_custom] Running controller.update()
2022-06-19 21:02:24 DEBUG (MainThread) [teslajsonpy.controller] Get vehicles. Force: False Time: 16 Interval 60
2022-06-19 21:02:24 DEBUG (MainThread) [teslajsonpy.controller] 16553: online. Polling policy: normal. Update state: normal. Since last park: 163. Since last wake_up: 161. Idle interval: 600. shift_state: None sentry: False climate: False, charging: Stopped
2022-06-19 21:02:24 DEBUG (MainThread) [teslajsonpy.controller] 16553: Skipping update with state online. Polling: True. Last update: 15 ago. Last parked: 163 ago. Last wake_up 161 ago.
2022-06-19 21:02:24 DEBUG (MainThread) [custom_components.tesla_custom] Finished fetching tesla_custom data in 0.002 seconds (success: True)
2022-06-19 21:02:34 DEBUG (MainThread) [custom_components.tesla_custom] Running controller.update()
2022-06-19 21:02:34 DEBUG (MainThread) [teslajsonpy.controller] Get vehicles. Force: False Time: 26 Interval 60
2022-06-19 21:02:34 DEBUG (MainThread) [teslajsonpy.controller] 16553: online. Polling policy: normal. Update state: normal. Since last park: 173. Since last wake_up: 171. Idle interval: 600. shift_state: None sentry: False climate: False, charging: Stopped
2022-06-19 21:02:34 DEBUG (MainThread) [teslajsonpy.controller] 16553: Skipping update with state online. Polling: True. Last update: 25 ago. Last parked: 173 ago. Last wake_up 171 ago.
2022-06-19 21:02:34 DEBUG (MainThread) [custom_components.tesla_custom] Finished fetching tesla_custom data in 0.002 seconds (success: True)
```
I see now that it appears to be Tesla's API returning a not yet updated state for `is_auto_conditioning_on`.

**Comment by alandtse** [comment]
Yah, we're reliant on Tesla reporting the correct state as we treat the API as a source of truth. We're actually keyed on [`is_climate_on`](https://github.com/zabuldon/teslajsonpy/blob/dev/teslajsonpy/homeassistant/climate.py#L61).

Right now it does this:
1. Issue command, if command reports success, manually update the state to match result of command. https://github.com/zabuldon/teslajsonpy/blob/dev/teslajsonpy/homeassistant/climate.py#L132-L138
2. Immediately force an API refresh on climate to update all sensors. This is where it's breaking because of the bad API response. https://github.com/zabuldon/teslajsonpy/blob/dev/teslajsonpy/homeassistant/climate.py#L146

I'm hesitant of changing the behavior. Perhaps we add a delay to 2 so it gives time for the API to fully update, but I'm not sure how long we should wait.

**Comment by andjo** [comment]
I don't know all the implications of step 2, but would it perhaps be an option to skip that step, or at least don't ask the API for a full update of everything. Just assume that the state now is what it is supposed to be for `is_auto_conditioning_on` and `is_climate_on` (provided that the API call didn't return an error of course). Or are there other values of interest that also can be updated by the call to auto_conditioning_start?

I notice while doing `vehicle_data` request on the side (with postman) just after turning climate on/off that the `is_climate_on` status is often updated before `is_auto_conditioning_on`.

**Comment by alandtse** [comment]
Multiple entities are linked to climate. Since you can only query entire sections, we update everything that may rely on the section.
